tests/unit/lms/services/h_api_test.py

```diff
@@ -187,44 +162,40 @@ def test_it_raises_HAPIError_for_other_http_errors(
         # It records the requests exception that caused the HAPIError.
         assert exc_info.value.__cause__ == exception
 
-    def test_other_exceptions_are_raised_normally(self, h_api, requests):
+    def test__api_request_raises_other_exceptions_normally(self, h_api, requests):
         requests.request.side_effect = OSError()
 
         with pytest.raises(OSError):
             h_api._api_request(sentinel.method, "dummy-path")
 
-    def _assert_called_requests_with(self, requests, **extras):
-        values = {
-            "method": Any(),
-            "url": Any(),
-            "auth": Any(),
-            "timeout": Any(),
-            "headers": Any(),
-        }
+    @pytest.fixture
+    def h_user(self):
+        return HUser(
+            username=sentinel.username,
+            display_name=sentinel.display_name,
+            authority="TEST_AUTHORITY",
+        )
 
-        values.update(extras)
+    @pytest.fixture
+    def h_api(self, pyramid_request):
+        return HAPI(sentinel.context, pyramid_request)
 
-        requests.request.assert_called_once_with(**values)
```

Comment:
I'm not a fan of test helpers like `_assert_called_requests_with()` except when really needed. Certainly not when they can easily be replaced with standard mock library asserts, as in this case. They make tests harder to understand (you now need to scroll down to this undocumented test helper method and understand what it does, before you can understand the test that uses). They also complicate test code, and clever test code (usually added in the name of avoiding test code duplication) often results in buggy tests. If not careful they can easily end up coupling tests together in undesirable ways too.